File: 1_ C/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int main()
{
    int owed;
    do
    {
        owed = get_int("Change owed: ");
    }
    while (owed < 0);

    int coin_types[] = {25, 10, 5, 1};
    int coins = 0;

    for (int i = 0; i < 4; i++)
    {
        int cur_coin = coin_types[i];
        while (owed >= cur_coin)
        {
            owed -= cur_coin;
            coins++;
        }
    }
    printf("%i\n", coins);
}

File: 6_ Python/sentimental-cash/cash.py
def main():
    while True:
        try:
            owed = float(input("Change: "))*100  # cents
            assert owed >= 0
            break
        except (ValueError, AssertionError):
            pass

    owed = int(owed)
    print(get_coins(owed))


def get_coins(change: int):
    coins = 0
    coin_types = [25, 10, 5, 1]
    for coin in coin_types:
        while change >= coin:
            coins += 1
            change -= coin
    return coins


if __name__ == "__main__":
    main()
